app/models/mixins/external_url_mixin.rb

```diff
@@ -0,0 +1,12 @@
+module ExternalUrlMixin
+  extend ActiveSupport::Concern
+
+  included do
+    has_many :external_urls, :as => :resource, :dependent => :destroy
+  end
+
+  def external_url=(url, user = User.current_user)
+    external_urls.where(:user => user).destroy_all
+    ExternalUrl.create!(:resource => self, :url => url, :user => user)
```

Comment:
minor but this leaves the external_urls cache inconsistent.  This imght be better...

```ruby
external_urls.where(:user => user).destroy_all
external_urls.create!(:url => url, :user => user)
```

Reply:
Will do.